### ISSUE TITLE
Inheritance and property overrides

### ISSUE THREAD

**Issue by romgerman** [GillianPerard/typescript-json-serializer#116]
Hey. So the issue is this

```js
@Serializable()
class BaseClass {
    @JsonProperty('ID')
    Id: number;
}

@Serializable()
class ChildClass extends BaseClass {
    @JsonProperty('Item_ID')
    Id: number;
}
```

```json
{
    "Item_ID": 12
}
```

```js
const data = deserialize(json, ChildClass);
console.log(data.Id); // undefined
```

If I try to override json property name in inherited class I get undefined value because it tries to find `ID` and not `Item_ID` although I parse `ChildClass` and not `BaseClass`.

**Comment by GillianPerard** [comment]
Hi! Ok I think when I merge the maps I used to define an object, the baseClass properties are priority.

Let me check if I'm true.

**Comment by GillianPerard** [comment]
I fixed the problem in the new 2.5.3 version 😊

Thank you again for your issue!
